mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -610,7 +610,7 @@ struct InitParamsXDL : InitParams, Serializable<InitParamsXDL> {
       : InitParams{mPerBlock, nPerBlock, kPerBlock}, gemmMPerWave(mPerWave),
         gemmNPerWave(nPerWave), gemmKPack(kPack),
         gemmAThreadCopyMoreGemmK(aThreadCopyMoreGemmK),
-        gemmBThreadCopyMoreGemmKPack(bThreadCopyMoreGemmKPack) {}
+        gemmBThreadCopyMoreGemmKPack(bThreadCopyMoreGemmKPack), gemmKBlocks(1LL) {}
```

Comment:
@jerryyin Shouldn't something like this go on the derived parameters? I could've sworn InitParams is supposed to be in 1:1 correspondence with the perf config

Reply:
@jerryyin Just to understand this amended requirement correctly: it seems we want to make `InitParams*` only contain values from `--perf_config`, therefore derived values such as `kblocks` shall be kept in a different structure.

Then, some subsequent questions for discussion. I'm open to ideas and can act accordingly.

- in which structure shall this value be kept? right now it seems there are `DerivedParams` and `DerivedOutParams`. Should we put this field in one of them, or create a new structure?

- what should be the interface to compute this derived parameter? right now we have `calculateInputDerivedParams` and `calculateOutputDerivedParams`, and `populateParamsXDL.obtainTuningParameters` is the main interface to compute these derived parameters.

Reply:
https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/060a944e06eb4395cb2d648906fb9ff4e9c11b20/mlir/lib/Dialect/MIOpen/Tuning/GridwiseGemmParams.cpp#L229-L233

In this function signature, we use the `InitParamsXDL` to derive the rest of parameters such that:
 - `gemmADerivedParam` store gemmA related derived parameters only
 - `gemmBDerivedParam` store gemmB related derived parameters only
 - `gemmCDerivedParam` store gemmC related derived parameters only

So it really depends on whether this parameter belong to a gemm argument or output. If not, then you'll likely need to follow on how `blockSize` or `gridSize` get passed along in `obtainTuningParameters()` routine.

Reply:
@jerryyin please take a look at the latest commit. `kblocks` is not really tied to a specific operand so I'm making it a standalone value at the moment.